front/lib/api/assistant/conversation.ts

```diff
@@ -365,17 +385,21 @@ export async function getConversationWithoutContent(
   });
 
   if (!conversation) {
-    return null;
+    return new Err(new ConversationNotFoundError());
   }
 
-  return {
+  if (!(await canAccessConversation(auth, conversation))) {
```

Comment:
This defeats the purpose of getConversationWithoutContent whose sole purpose is to read a conversation fast without loading messages.

Eg. in the praticipants endpoint we call it and then fetch the participants but now that call also fetch the participants which means we fetch the participants twcie

Reply:
Yes, this was a conscious tradeoff as described in the PR risks. I think it's the least damageable option

I went through all the callers of `getConversationWithoutContent` and the permission check was legit every time, except for the participants endpoint. So we could move the check out but IMO would make the code more brittle because current stats show the check must be done almost all the time (1 exception out of ~15 IIRC)

Basically we're putting mental load on the callers and the check will likely be missed at some point. Also, having a Result here seemed more robust and allowed some code drying